workflow/executor/executor.go

```diff
@@ -77,6 +79,12 @@ type WorkflowExecutor struct {
 	// list of errors that occurred during execution.
 	// the first of these is used as the overall message of the node
 	errors []error
+
+	// current progress which is synced every `annotationPatchTickDuration` to the pods annotations.
+	progress string
```

Comment:
minor - could be `Progress` type?

Reply:
I can make a PR to change this, probably tomorrow 👍